packages/@aws-cdk/aws-s3-deployment/lib/lambda/index.py

```diff
@@ -115,8 +113,13 @@ def cfn_error(message=None):
 #---------------------------------------------------------------------------------------------------
 # populate all files from s3_source_zips to a destination bucket
 def s3_deploy(s3_source_zips, s3_dest, user_metadata, system_metadata, prune):
-    # create a temporary working directory
-    workdir=tempfile.mkdtemp()
+    # create a temporary working directory in /tmp or if enabled an attached efs volume
+    if ENV_KEY_MOUNT_PATH in os.environ:
+        workdir = os.getenv(ENV_KEY_MOUNT_PATH) + "/" + str(uuid4())
+        os.mkdir(workdir)
+    else:
+        workdir = tempfile.mkdtemp()
```

Comment:
Since this `workdir` is now persisted in the EFS, we need to make sure to cleanup after ourselves. i.e put the deletion of it in a finally clause instead of just:

https://github.com/aws/aws-cdk/blob/f58cf3c95eb32e9a4dc797665160e1b508ace2e1/packages/%40aws-cdk/aws-s3-deployment/lib/lambda/index.py#L144-L146

Reply:
Make sense. Will update this.